crates/accelerate/src/permutation.rs

```diff
@@ -111,10 +170,30 @@ fn _get_ordered_swap(py: Python, permutation_in: PyArrayLike1<i64>) -> PyResult<
     Ok(get_ordered_swap(&view).to_object(py))
 }
 
+/// Decompose a SWAP pattern into a series of SWAP gate indices to implement them.
+/// For example, let the pattern be [1, 2, 3, 4, 0, 6, 5], which contains the two cycles
+/// [1, 2, 3, 4, 0] and [6, 5]. These can then be implemented with the SWAPs
+/// [(0, 3), (1, 2), (0, 4), (1, 3)], respectively [(6, 5)].
+/// If ``invert`` is True, reverse the indices before computing the SWAPs.
+#[pyfunction]
+#[pyo3(signature = (pattern, invert_order=true))]
+fn _pattern_to_swaps(
+    py: Python,
+    pattern: PyArrayLike1<i64>,
```

Comment:
Naive question, why did you pick i64 instead of i32? I also wonder if there is something like `PyArrayViewLike1` you could use where you directly work with references instead of manually getting the reference in line 186. I am not 100% sure if that is better but it feels like it could save a few steps.

Reply:
Naive answer: because the other functions ported before also used `i64`. This is also the answer to the type choice, but we should check if using `PyArray...` would be better for all these -- let's sync offline 🙂

Reply:
Hi @ElePT and @Cryoris -- I chose `i64` for `_get_ordered_swap` because it's the default integer type for Numpy arrays on Linux. There is some cost for converting from other types. On my machine, if I run the Rust version of `_get_ordered_swap` 10k times I get:

| Input type | Time (seconds) |
|---------------|-------------|
| Numpy array (int32) | 0.04141 |
| Numpy array (int64) | 0.00921 |
| Numpy array (uint32)	| 0.04026 |
| Numpy array (uint64)	| 0.04066 |
| Python list | 0.01595 |

I don't know how much this matters, because building the actual circuit will probably always be the bottleneck.